src/placement/jump_map_versions.c

```diff
@@ -439,11 +443,36 @@ dom_reset_full(struct pool_domain *dom, uint8_t *dom_bits, uint8_t *tgts_used,
 		clrbit(tgts_used, &dom->do_targets[i] - root->do_targets);
 }
 
+static bool
+dom_tgts_are_avaible(struct pool_domain *dom, uint32_t allow_status, uint32_t allow_version)
+{
+	int i;
+
+	for (i = 0; i < dom->do_target_nr; i++) {
+		struct pool_target *tgt;
+		uint32_t status;
+
+		tgt = &dom->do_targets[i];
+		status = tgt->ta_comp.co_status;
+		if (tgt->ta_comp.co_status == PO_COMP_ST_DOWN) {
+			if (tgt->ta_comp.co_fseq > allow_version)
+				status = PO_COMP_ST_UPIN;
+		} else if (tgt->ta_comp.co_status == PO_COMP_ST_UP) {
+			if (tgt->ta_comp.co_in_ver > allow_version)
+				status = PO_COMP_ST_DOWNOUT;
```

Comment:
this case seems also possible be PO_COMP_ST_NEW?
but probably does not affect result

Reply:
oh, this function is to "correct" the status first by version first, so only DOWN and UP are needed to be "corrected" by version. NEW targets are not needs to be changed, it can be used to compare the status anyway.